pkg/licensing/normalize_test.go

```diff
@@ -1,13 +1,236 @@
-package licensing_test
+package licensing
 
 import (
 	"testing"
 
 	"github.com/stretchr/testify/assert"
-
-	"github.com/aquasecurity/trivy/pkg/licensing"
 )
 
+// All map keys must be standardized to be matched
+// (uppercase, no common suffixes, standardized version, etc.)
+func TestMap(t *testing.T) {
+	for key := range mapping {
+		t.Run(key, func(t *testing.T) {
+			standardized := standardizeKeyAndSuffix(key)
+			assert.Equal(t, standardized.License, key)
+		})
+	}
+}
+
+func TestNormalize(t *testing.T) {
+	tests := []struct {
+		licenses      []string
+		normalized    string
+		normalizedKey string
+	}{
+		{
+			[]string{
+				"  the apache license  ",
+				"  the\tapache  \r\nlicense  \r\n ",
+				" apache ",
+				"ApacheLicence",
+				"ApacheLicense",
+				"al-2",
+				"al-v2",
+				"al2",
+				"alv2",
+				"apache - v 2.0",
+				"apache - v. 2.0",
+				"apache - ver 2.0",
+				"apache - version 2.0",
+				"apache 2",
+				"apache 2.0",
+				"apache license (2.0)",
+				"apache license (v. 2)",
+				"apache license (v. 2.0)",
+				"apache license (v2)",
+				"apache license (v2.0)",
+				"apache license (version 2.0)",
+				"apache license 2",
+				"apache license 2.0",
+				"apache license v2",
+				"apache license v2.0",
+				"apache license version 2",
+				"apache license version 2.0",
+				"apache license",
+				"apache license, 2.0",
+				"apache license, asl version 2.0",
+				"apache license, version 2",
+				"apache license, version 2.0 (http://www.apache.org/licenses/license-2.0)",
+				"apache license, version 2.0",
+				"apache license,version 2.0",
+				"apache license,version-2.0",
+				"apache license-2.0",
+				"apache public 2.0",
+				"apache public license 2.0",
+				"apache public license-2.0",
+				"apache public-2",
+				"apache public-2.0",
+				"apache software license (apache-2.0)",
+				"apache software license - version 2.0",
+				"apache software license 2.0",
+				"apache software license, version 2",
+				"apache software license, version 2.0",
+				"apache software-2.0",
+				"apache v 2.0",
+				"apache v. 2.0",
+				"apache v2",
+				"apache v2.0",
+				"apache ver 2.0",
+				"apache ver. 2.0",
+				"apache version 2.0",
+				"apache version 2.0, january 2004",
+				"apache version-2",
+				"apache version-2.0",
+				"apache",
+				"apache, 2",
+				"apache, v2.0",
+				"apache, version 2",
+				"apache, version 2.0",
+				"apache-2",
+				"apache-2.0",
+				"apache-licence",
+				"apache-license",
+				"apache-licensed",
+				"apache-licensed",
+				"asf 2.0",
+				"asl 2",
+				"asl, version 2",
+				"asl2.0",
+				"the apache license",
+				"the apache license",
+			},
+			"Apache-2.0",
+			"Apache-2.0",
+		},
+		{
+			[]string{
+				"Apache+",
+			},
+			"Apache-2.0+",
+			"Apache-2.0",
```

Comment:
Can you add field names for better readability in these tests?
e.g.:
```suggestion
			licenses: []string{
				"Apache+",
			},
			normalized:    "Apache-2.0+",
			normalizedKey: "Apache-2.0",
```

Reply:
Field names have been added in last commit.